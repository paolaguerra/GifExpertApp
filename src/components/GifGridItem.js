import PropTypes from 'prop-types';

export const GifGridItem = ({title, url, id}) => {

  return (
    
    <div className="card">
        <img className="card-img-top" src={url} alt={title} />
        <div class="card-body">
              <p class="card-text">{title}</p>
            </div>
    </div>
  )
}


        {/* <div class="card" style="width: 18rem;">
          <img src="..." class="card-img-top" alt="..." />
            <div class="card-body">
              <p class="card-text">Some quick example text to build on the card title and make up the bulk of the card's content.</p>
            </div>
        </div> */}

GifGridItem.prototype = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}