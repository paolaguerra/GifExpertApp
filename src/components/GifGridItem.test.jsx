import { render, screen } from "@testing-library/react"
import { GifGridItem } from "./GifGridItem"

describe('Prueba en <GifGridItem />', () => { 

    const title = 'Ahsoka';
    const url = 'https://star-wars.com/ahsoka.jpg';

    test('debe de hacer match con el snapchot', () => { 

        const {container} = render(<GifGridItem title={title} url={url} />);
        expect(container).toMatchSnapshot();

     });

     test('debe de mostrar la imagen con el URL y el ALT indicado', () => { 

        render(<GifGridItem title={title} url={url} />);
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);

     });

     test('debe de mostrar el titulo en el componente', () => { 

        render(<GifGridItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();

      });

 });