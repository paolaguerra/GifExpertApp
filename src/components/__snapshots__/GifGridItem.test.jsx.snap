// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifGridItem /> debe de hacer match con el snapchot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Ahsoka"
      src="https://star-wars.com/ahsoka.jpg"
    />
    <p>
      Ahsoka
    </p>
  </div>
</div>
`;
